src/Control/Director.php

```diff
@@ -841,28 +842,16 @@ public static function is_site_url($url)
 
     /**
      * Given a filesystem reference relative to the site root, return the full file-system path.
-     *
-     * @param string $file
-     *
-     * @return string
      */
-    public static function getAbsFile($file)
+    public static function getAbsFile(string $file): string
```

Comment:
Strong typing added here because the symfony `Path` class has strong typing - so we can't allow other types to come in anymore.

I've added similar typing changes in other methods across the PRs.